tools/azsdk-cli/Azure.Sdk.Tools.Cli/Program.cs

```diff
@@ -0,0 +1,39 @@
+using System.CommandLine;
+using Azure.Sdk.Tools.Cli.Commands;
+using Azure.Sdk.Tools.Cli.Contract;
+using Azure.Sdk.Tools.Cli.Services.Azure;
+using ModelContextProtocol.Protocol.Types;
+
+namespace Azure.Sdk.Tools.Cli;
+
+public sealed class Program
+{
+    public static async Task<int> Main(string[] args)
+    {
+        // create the default service collection which will make logging and commandfactory available for use
+        // The command factory itself will walk the assembly and discover any MCPTool implementors, and register them
+        // as commands.
+        var services = new ServiceCollection();
+        services.AddLogging(builder =>
+        {
+            builder.AddConsole();
+            builder.SetMinimumLevel(LogLevel.Debug);
+        });
+        services.AddSingleton<CommandFactory>();
+
+        // any registrations below here are only necessary for CLI mode
+        // the hosttool mode re-inits its own service provider
+        services.AddSingleton<IAzureService, AzureService>();
+
```

Comment:
Discover and register all of your tools here:
```suggestion

        RegisterTools(services);
```

where RegisterTools can allow each tool to register additional services, or just add all tools by base class from assembly lookup.

```
   HellowWorldTool.Register(services);
   
   vs
   
   AppDomain.CurrentDomain
                         .GetAssemblies()....
```